sources/xfscli/src/bin/fsck_xfs.rs

```
@@ -0,0 +1,169 @@
+use std::os::unix::fs::FileTypeExt;
+use std::path::Path;
+use std::process::exit;
+use std::process::Command;
+use std::str;
+
+use argh::FromArgs;
+
+use xfscli::perform_xfs_repair;
+use xfscli::{BLKID, MOUNT, UMOUNT};
+
+// This CLI is rewritten in Rust for fsck.xfs shell script
+// https://git.kernel.org/pub/scm/fs/xfs/xfsprogs-dev.git/tree/fsck/xfs_fsck.sh
+// since there isn't a shell on Bottlerocket hosts.
+
+/// check and repair a XFS file system
+#[derive(FromArgs)]
+struct Args {
+    /// auto repair the file system
+    #[argh(switch, short = 'a')]
+    auto_repair: bool,
+    /// auto check all file systems
+    #[argh(switch, short = 'A')]
+    auto_check: bool,
+    /// check the root filesystem as well when -A flag is set
+    #[argh(switch, short = 'p')]
+    check_root_fs: bool,
+    /// force repair filesystem
+    #[argh(switch, short = 'f')]
+    force: bool,
+    /// repair filesystem automatically
+    #[argh(switch, short = 'y')]
+    repair: bool,
+
+    /// mount-point | block-device
+    #[argh(positional)]
+    target: Option<String>,
+}
+
+fn main() {
+    let args: Args = argh::from_env();
+    let auto = args.auto_check || args.auto_repair || args.check_root_fs;
+    let force = args.force;
+    let repair = args.repair;
+
+    if let Some(target) = &args.target {
+        if !Path::new(target).exists() {
+            println!("{} does not exist", target);
+            exit(8);
+        }
+
+        if force {
+            // Preform xfs_repair
+            println!("Performing XFS repair..");
+            let mut repair_exit_code = perform_xfs_repair(vec!["-e", target]);
+
+            // Clear the log, mount and unmount the XFS file system
+            // before xfs_repair
+            if repair_exit_code == 2 && repair {
+                println!("Replaying log for {}", target);
+                let mkdir_result = fs::create_dir_all("/tmp/repair_mnt");
+                if mkdir_result.is_err() {
+                    exit(1);
```

Do you need to print out or give more details about the error? You could return a readable error and `Err` and the exit code will still be `1`